backend/helper_zero/views/auth.py

```diff
@@ -1,13 +1,20 @@
+import os
 from rest_framework import status, viewsets
 from rest_framework.response import Response
+from google.oauth2 import id_token
+from google.auth.transport import requests
 
 from helper_zero.models import Organization, DonationRequest
 from helper_zero.serializers import OrganizationSerializer, DonationRequestSerializer
 
 class AuthView(viewsets.ViewSet):
   def list(self, request):
     auth_token = request.query_params.get('auth_token')
-    org = Organization.objects.filter(auth_token=auth_token).first()
+    id_info = id_token.verify_oauth2_token(auth_token, requests.Request(), os.environ['GOOGLE_CLIENT_ID'])
+    if id_info['iss'] not in ['accounts.google.com', 'https://accounts.google.com']:
+      return Response(status=status.HTTP_401_UNAUTHORIZED)
+    user_id = id_info['sub']
+    org = Organization.objects.filter(auth_user_id=user_id).first()
```

Comment:
What necessitates the usage of `.first()`? Does that imply there could be more than one entry?

Reply:
by default, objects.filter returns a list. there should never be more than one item in that list (i can add validation later, but for now, just want to get a working model up)

Reply:
Hm ok, could be good to just check the length of the list immediately and return a 4XX code if more than one but yeah. Maybe add that as a future TODO on Trello